### PR TITLE
Add Nial Lexer (feature branch)

### PR DESCRIPTION
**Pull request by razetime**
Cherry picked commits, based on the instructions from #1803 [here](https://github.com/rouge-ruby/rouge/pull/1803#issuecomment-1170721284).

### CODE REVIEW

File: spec/lexers/nial_spec.rb [hunk 1]
```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+describe Rouge::Lexers::Nial do
```
Comment (tancnle) [inline spec/lexers/nial_spec.rb:4]: **suggestion:** Would it be possible to fix up the indentation on this spec as well 🙏🏼 ?

Comment (razetime) [reply spec/lexers/nial_spec.rb:4]: I've changed it to 2 spaces.

File: spec/visual/samples/nial [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+'
+'
+
+% Non-keyword names ;
+thing here there NotAKeyword
+
+% ops and punctuation ;
+/[sum, tally]
+
+(adasd +)
```
Comment (tancnle) [inline spec/visual/samples/nial:121]: **issue:** Could we add an end-of-line to this file? The CI is failing.

Comment (razetime) [reply spec/visual/samples/nial]: I've added the end-of-line to it.

**Comment by razetime** [comment]
The CI is still failing. What formatter can I use to fix my code style? rubocop reports no problems.

File: lib/rouge/demos/nial [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     5 / 5 10 0 8
+# OUTPUT:
+1. 0.5 ?div 0.625
+
+     % You can also define custom faults using 'fault': ;
+     fault 'this is an error'
+# OUTPUT:
+this is an error
+
+
```
Comment (tancnle) [inline lib/rouge/demos/nial:37]: **suggestion:** `linelint` does not like empty lines. Could we remove these?

File: lib/rouge/lexers/nial.rb [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      
+      state :str do
+        rule %r/''/, Str::Escape
+        rule %r/[^']+/, Str::Single
+        rule %r/'|$/, Str::Single, :pop!
+      end
+    end
+  end
+end
+
```
Comment (tancnle) [inline lib/rouge/lexers/nial.rb:167]: **suggestion:** `linelint` does not like empty lines. Could we remove these?

Comment (razetime) [reply lib/rouge/lexers/nial.rb]: Removed the lines

File: lib/rouge/demos/nial (hunk 3 above)
Comment (razetime) [reply lib/rouge/demos/nial]: Removed the lines

**Comment by tancnle** [comment]
This is exciting @razetime. Thank you for your wonderful support on this. Let's ship it 🚀

**Pull request merged**